tests/page_test.go

```diff
@@ -1873,3 +1882,174 @@ func TestPageGetAttributeEmpty(t *testing.T) {
+	tb := newTestBrowser(t, withHTTPServer())
+	tb.withHandler("/home", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, `
+		<html>
+			<head></head>
+			<body>
+				<script type="module">
+					await ping();
+					async function ping() {
+						await fetch('/ping?h=2kq2lo6n06');
+						await fetch('/ping?h=ej0ypprcjk');
+					}
+				</script>
+			</body>
+		</html>`)
+		require.NoError(t, err)
+	})
+	tb.withHandler("/ping", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprintf(w, `pong`)
+		require.NoError(t, err)
+	})
+
+	ignoreURLs := map[string]interface{}{
+		tb.url("/home"):        nil,
+		tb.url("/favicon.ico"): nil,
+	}
+
+	tests := []struct {
+		name string
+		fun  string
+		want string
+	}{
+		{
+			name: "single_page.on",
+			fun: `page.on('metric', (msg) => {
+				msg.groupURLTag({
+				  groups: [
+						{url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-1'},
+					]
+				});
+			});`,
+			want: "ping-1",
+		},
+		{
+			name: "multi_groupURLTag",
+			fun: `page.on('metric', (msg) => {
+				msg.groupURLTag({
+				  groups: [
+						{url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-1'},
+					]
+				});
+				msg.groupURLTag({
+					groups: [
+						  {url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-2'},
+					  ]
+				  });
+			});`,
+			want: "ping-2",
+		},
+		{
+			name: "multi_groupURLTag_page.on",
+			fun: `page.on('metric', (msg) => {
+				msg.groupURLTag({
+				  groups: [
+						{url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-1'},
+					]
+				});
+				msg.groupURLTag({
+					groups: [
+						  {url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-2'},
+					  ]
+				  });
+			});
+			page.on('metric', (msg) => {
+				msg.groupURLTag({
+				  groups: [
+						{url: /^http:\/\/127\.0\.0\.1\:[0-9]+\/ping\?h=[0-9a-z]+$/, name:'ping-3'},
+					]
+				});
+			});`,
+			want: "ping-3",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			var foundAmended int
+			var foundUnamended int
+			var foundMu sync.RWMutex
+
+			done := make(chan bool)
+
+			samples := make(chan k6metrics.SampleContainer)
+			go func() {
+				defer close(done)
+				for e := range samples {
+					ss := e.GetSamples()
+					for _, s := range ss {
+						// At the moment all metrics that the browser emits contains
+						// both a url and name tag on each metric.
+						u, ok := s.TimeSeries.Tags.Get("url")
+						assert.True(t, ok)
+						n, ok := s.TimeSeries.Tags.Get("name")
+						assert.True(t, ok)
+
+						// The name and url tags should have the same value.
+						assert.Equal(t, u, n)
+
+						// If the url is in the ignoreURLs map then this will
+						// not have been matched on by the regex, so continue.
+						if _, ok := ignoreURLs[u]; ok {
+							foundMu.Lock()
+							foundUnamended++
+							foundMu.Unlock()
```

Comment:
I'd use the [atomic package](https://pkg.go.dev/sync/atomic) here.

Reply:
Sounds good, updated in a3fefb17d024ec9038340cee0540cc02b98a0183